src/topology/metric_space/basic.lean

```diff
@@ -863,6 +863,14 @@ lemma closed_ball_Icc {x r : ℝ} : closed_ball x r = Icc (x-r) (x+r) :=
 by ext y; rw [mem_closed_ball, dist_comm, real.dist_eq,
   abs_sub_le_iff, mem_Icc, ← sub_le_iff_le_add', sub_le]
 
+-- This lemma could go somewhere else, but I did not find a better fit for it.  Feel
+-- free to suggest where it may be added.  Here, all the needed imports are available!
+lemma mem_Icc_iff_abs_le {R : Type*} [linear_ordered_add_comm_group R] {x y z : R} :
+  abs (x - y) ≤ z ↔ y ∈ Icc (x - z) (x + z) :=
+abs_le.trans ⟨
+  λ h, ⟨sub_le.mp h.2, neg_le_sub_iff_le_add.mp h.1⟩,
+  λ hy, ⟨neg_le_sub_iff_le_add.mpr hy.2, sub_le.mp hy.1⟩⟩
```

Comment:
Even shorter:
```suggestion
abs_le.trans $ (and_comm _ _).trans $ and_congr sub_le neg_le_sub_iff_le_add
```

Reply:
The shorter the proofs get, the more trivial the result appears...  Thanks!